sched/semaphore/spinlock.c

```diff
@@ -384,4 +384,77 @@ void spin_clrbit(FAR volatile cpu_set_t *set, unsigned int cpu,
 }
 #endif
 
+void fair_spinlock_list_init(fair_spinlock_list_t *lock_list)
+{
+  lock_list->lock = SP_UNLOCKED;
+  list_initialize(&lock_list->list);
+}
+
+void fair_spinlock_init(fair_spinlock_t *lock)
+{
+  lock->lock = SP_LOCKED;
+  lock->holder = this_task();
+  list_initialize(&lock->node);
+}
+
+void fair_spin_lock(FAR fair_spinlock_list_t *lock_list,
```

Comment:
Please include the function documentation decoration describing the function and its parameters.